### PR TITLE
Top course for teachers

### PR DESCRIPTION
**Pull request by Erin007**
Students currently see the course/script in which they most recently have progress as a `StudentTopCourse` component above all of their `CourseCard`s.  This change gives teachers a `TopCourse`, too. 
teacher-topcourse

- Renamed StudentTopCourse to TopCourse
- Used TopCourse in RecentCourses for everyone, not just students 
- Pass top_course data to TeacherHomepage
- Wrote test for TopCourse 
- Updated tests for RecentCourses, StudentHomepage and TeacherHomepage

### CODE REVIEW

**Comment by Erin007** [comment]
@tanyaparker

File: apps/src/templates/studioHomepages/TopCourse.jsx [hunk 1]
```diff
@@ -66,9 +66,9 @@ const styles = {
   },
 };
 
-// While this is named a StudentTopCourse, it really refers to the most recent course or script in which the student has progress.
+// While this is named TopCourse, it really refers to the most recent course or script in which the student or teacher has progress.
```
Comment (breville) [inline apps/src/templates/studioHomepages/TopCourse.jsx]: nit: long line

File: apps/test/unit/templates/studioHomepages/RecentCoursesTest.js [hunk 2]
```diff
@@ -7,149 +7,210 @@ import SetUpCourses from '@cdo/apps/templates/studioHomepages/SetUpCourses';
...
+            linkToLesson={topCourse.linkToLesson}
+          />
+          <div key={0}>
+            <CourseCard
+              title={courses[0].title}
+              description={courses[0].description}
+              link={courses[0].link}
+              isRtl={false}
+            />
+            <div>.</div>
```
Comment (breville) [inline apps/test/unit/templates/studioHomepages/RecentCoursesTest.js:112]: Given how explicit this declaration of the DOM is, down to a period, I wonder if this form of matching is a little brittle?

File: dashboard/app/controllers/home_controller.rb [hunk 3]
```diff
@@ -86,31 +86,24 @@ def init_homepage
       @sections = current_user.sections.map(&:summarize)
       @student_sections = current_user.sections_as_student.map(&:summarize)
 
-      # Students will receive a @student_top_course for their primary script,
-      # so we don't want to include that script (if it exists) in the regular
-      # lists of recent scripts.
-      exclude_primary_script = current_user.student?
+      # Students and teachers will receive a @top_course for their primary script, so we don't want to include that script (if it exists) in the regular lists of recent scripts.
```
Comment (breville) [inline dashboard/app/controllers/home_controller.rb]: nit: long line

File: dashboard/app/controllers/home_controller.rb [hunk 4]
```diff
@@ -86,31 +86,24 @@ def init_homepage
...
+      exclude_primary_script = current_user
       @recent_courses = current_user.recent_courses_and_scripts(exclude_primary_script)
 
-      if current_user.teacher?
-        @sections = current_user.sections.map(&:summarize)
+      script = current_user.primary_script
+      if script
```
Comment (breville) [inline dashboard/app/controllers/home_controller.rb:96]: Seems like we should test these changes in `home_controller_test.rb`.

**Comment by breville** [comment]
Have you been able to test that everything works fine when there is no top course?

File: dashboard/app/controllers/home_controller.rb [hunk 5]
```diff
@@ -86,31 +86,24 @@ def init_homepage
       @sections = current_user.sections.map(&:summarize)
       @student_sections = current_user.sections_as_student.map(&:summarize)
 
-      # Students will receive a @student_top_course for their primary script,
-      # so we don't want to include that script (if it exists) in the regular
-      # lists of recent scripts.
-      exclude_primary_script = current_user.student?
+      # Students and teachers will receive a @top_course for their primary script, so we don't want to include that script (if it exists) in the regular lists of recent scripts.
+      exclude_primary_script = current_user
```
Comment (Bjvanminnen) [inline dashboard/app/controllers/home_controller.rb]: we already know current_user be truthy here (see line 81).

also, nit: i'd prefer to see line 89 wrapped

Comment (Bjvanminnen) [reply dashboard/app/controllers/home_controller.rb]: I also wonder if this API of passing a top course and other courses still makes sense in a world where both student and teacher use top course. I wonder if we can simplify things. Maybe pass a list of all courses, and an index of which one is the "top" course, and then let the client determine what to do with these.

**Comment by Erin007** [comment]
When there isn't a top course, the homepage just shows a SetUpMessage, same as before with students.

**Pull request merged**